animation/geometry.h

```diff
@@ -367,12 +367,143 @@ namespace animation
                     dimension::get <1> (p) >= y1 &&
                     dimension::get <1> (p) <= y2);
         }
+
+        /**
+         * Vector4DModel<T>:
+         *
+         * A detached 2D point or vector in space for a given data
+         * type T. This is a structure of two values.
+         *
+         * PointModel implements the Dimension trait, meaning that it
+         * can be used with functions in the animation::geometry::dimension
+         * namespace.
+         */
+        template <typename T>
+        struct Vector4DModel
+        {
+            Vector4DModel (T x, T y, T z, T w) noexcept :
+                x (x),
+                y (y),
+                z (z),
+                w (w)
+            {
+            }
+
+            Vector4DModel () noexcept :
+                x (0),
+                y (0),
+                z (0),
+                w (0)
+            {
+            }
+
+            Vector4DModel (Vector4DModel const &v) noexcept :
+                x (v.x),
+                y (v.y),
+                z (v.z),
+                w (v.w)
+            {
+            }
+
+            void swap (Vector4DModel &a, Vector4DModel &b) noexcept
+            {
+                std::swap (a.x, b.x);
+                std::swap (a.y, b.y);
+                std::swap (a.z, b.w);
```

Comment:
Typo?

Reply:
Yikes, it was indeed. Thanks for spotting.